src/analyzer.rs

```diff
@@ -0,0 +1,169 @@
+use serde::{Deserialize, Serialize};
+use typed_builder::TypedBuilder;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum AnalyzerFeature {
+    Frequency,
+    Norm,
+    Position,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum AnalyzerCase {
+    Lower,
+    None,
+    Upper,
+}
+
+#[derive(Debug, Serialize, Deserialize, TypedBuilder, PartialEq)]
+#[builder(doc)]
+pub struct DelimiterAnalyzerProperties {
+    //The value will be used as delimiter to split text into tokens as specified
+    //in RFC 4180, without starting new records on newlines.
```

Comment:
Would you like this line to show up in documentation? Then please starts the doc comment with `///` instead of `//`. 

Thank you!

Reply:
Done